semantic_conventions/client.yaml

```
@@ -0,0 +1,32 @@
+groups:
+  - id: client
+    prefix: client
+    type: attribute_group
+    brief: These attributes may be used to describe the client - initiator of a network connection for events regarding sessions, connections, or bidirectional flow records.
+    attributes:
+      - id: address
+        type: string
+        brief: Client address - unix domain socket name, IPv4 or IPv6 address.
+        note: >
+          When communicating through intermediary, `client.address` SHOULD represent client address behind
+          any intermediaries (e.g. proxies) if it's available. Otherwise, it SHOULD represent immediate client peer address.
+        examples: ['/tmp/my.sock', '10.1.2.80' ]
+      - id: port
+        type: int
+        brief: 'Client port number'
+        examples: [65123]
+        note: >
+          When communicating through intermediary, `client.port` SHOULD represent client port behind
+          any intermediaries (e.g. proxies) if it's available. Otherwise, it SHOULD represent immediate client peer port.
+      - id: socket.address
+        type: string
+        brief: Immediate client peer address - unix domain socket name, IPv4 or IPv6 address.
+        examples: ['/tmp/my.sock', '127.0.0.1' ]
+        requirement_level:
+          recommended: Only if different than `client.address`.
+      - id: socket.port
+        type: int
+        brief: 'Immediate client peer port number'
+        examples: [35555]
+        requirement_level:
+          recommended: Only if different than `client.port`.
```

does this simplify things?

```suggestion
      - id: address
        type: string
        brief: Client address - unix domain socket name, IPv4 or IPv6 address.
        note: >
          When communicating through intermediary, `client.address` SHOULD represent client address behind
          any intermediaries (e.g. proxies) if it's available.
        examples: ['/tmp/my.sock', '10.1.2.80' ]
      - id: port
        type: int
        brief: 'Client port number'
        examples: [65123]
        note: >
          When communicating through intermediary, `client.port` SHOULD represent client port behind
          any intermediaries (e.g. proxies) if it's available.
      - id: socket.address
        type: string
        brief: Immediate client peer address - unix domain socket name, IPv4 or IPv6 address.
        examples: ['/tmp/my.sock', '127.0.0.1' ]
        requirement_level: recommended
      - id: socket.port
        type: int
        brief: 'Immediate client peer port number'
        examples: [35555]
        requirement_level: recommended
```

---

do you mean removing the requirement to populate  `client.socket.*` only when they don't match `client.* ` logical? 

If we want to avoid duplication, we'd have to define corresponding requirement level and condition everywhere. Let me  know if I messed the point :)

---

sorry, that big diff was super unclear, these are the changes I'm wondering about:

* `client.(address|port)` - remove "Otherwise, it SHOULD represent immediate client peer address."
* `client.socket.(address|port)` - remove condition on being different from `client.(address|port)`

it also sort of relates to @Oberon00's https://github.com/open-telemetry/opentelemetry-specification/pull/3402#discussion_r1173363136

> I think HTTP instrumentations that read IP information from the socket should also set only `server.socket.address`

---

We have those conditions to avoid duplication and also to have a consistent story for `client` and `server`.
I.e.
1. vendors and queries can always use `server|client.address`. These attributes are always present
2. To get a physical address,  check `server|client.socket.address`. When not provided, fall back to `server|client.address`

By removing these conditions, we'd require a different logical from vendor/user query:

1. To identify logical peer address, first check `server|client.address` and if absent, fall back to `server|client.socket.address`
2. To identify physical peer address use `server|client.socket.address` only

So currently we're optimizing querying by logical address, by removing conditions, we're making it harder to query. 

Let's discuss at HTTP semcomv call.

---

Discussed, and the current state looks good:
* `client.address` gives you the "best known" address for the client
* you can compare that against `client.socket.address` to know if it's more than just a direct peer (i.e. it's the "real" client address)